packages/runtime-common/indexer.ts

```diff
@@ -951,7 +947,15 @@ export class Batch {
 
   private async calculateInvalidations(id: string): Promise<string[]> {
     let invalidations = [id];
-    let childInvalidations = await this.client.cardsThatReference(id);
+    let childInvalidations: string[] = [];
+    let page = { number: -1, size: 10};
+    let results: string[];
+    do {
+      page.number++;
+      results = await this.client.cardsThatReference(id, page);
+      childInvalidations = [...childInvalidations, ...results];
+    } while(results.length === page.size);
```

Comment:
the paging should happen inside of `cardsThatReference()` callers should not have to know it's paged, that's an interior details of the `cardsThatReference()` method.